Leader Key/UserState.swift

```diff
@@ -17,11 +18,21 @@ final class UserState: ObservableObject {
     display = lastChar
     self.currentGroup = currentGroup
     self.isShowingRefreshState = isShowingRefreshState
+    self.navigationPath = nil
   }
 
   func clear() {
     display = nil
-    currentGroup = userConfig.root
+    currentGroup = nil
+    navigationPath = nil
     isShowingRefreshState = false
   }
+
+  func navigateToGroup(_ group: Group) {
+    if navigationPath == nil {
+      navigationPath = []
+    }
+    navigationPath?.append(group)
```

Comment:
```suggestion
    navigationPath.append(group)
```

We could even make `currentGroup` evaluate to `navigationPath.last` and skip some duplication